### ISSUE TITLE
Cherry-pick bot failing with `pushToNamedFork is not implemented in the v2 client`

### ISSUE THREAD

**Issue by killianmuldoon** [kubernetes/test-infra#30937]
**What happened**:
[k8s-infra-cherrypick-robot](https://github.com/k8s-infra-cherrypick-robot) is failing to do cherry-picks with the error `failed to push cherry-picked changes in GitHub: pushToNamedFork is not implemented in the v2 client`

An example of this error occurring is 

**What you expected to happen**:
Cherry-pick PR to be opened against correct branch

**How to reproduce it (as minimally and precisely as possible)**:
Comment the `/cherry-pick release-1.8` on PR https://github.com/kubernetes-sigs/cluster-api-provider-vsphere
/pull/2397
 
**Please provide links to example occurrences, if any**:
https://github.com/kubernetes-sigs/cluster-api-provider-vsphere
/pull/2397#issuecomment-1746652732
**Anything else we need to know?**:

**Comment by chrischdi** [comment]
Maybe related to #30924

**Comment by killianmuldoon** [comment]
@cjwagner just in case you have some idea if that change could have this impact.

**Comment by cjwagner** [comment]
Yeah that PR is definitely the cause of this. I didn't realize there was an unimplemented method in the v2 client. I'll push a fix ASAP (by tomorrow).

**Comment by andyzhangx** [comment]
this is not fixed yet, I could still get the same error.

**Comment by cjwagner** [comment]
The code should be fixed, it just needs to be deployed: https://github.com/kubernetes/test-infra/pull/30981
